### PR TITLE
python312Packages.gpgme: fix build

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

Noticed in #281106

I tried building python312Packages.python-swiftclient which now gets past gpgme but fails at a different package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by dotlambda** [comment]
Do you see a way we can use `fetchpatch`?

**Comment by SuperSandro2000** [comment]
> Do you see a way we can use `fetchpatch`?

I had that initially but I don't think https://dev.gnupg.org/D545?download=true is stable and I am not sure if the link it redirects to (https://files.gnupg.net/file/data/rbaxpebkyfbchg73llt2/PHID-FILE-bmkoy3pj2jjjwf7w6l54/D545.diff) is stable and permanent.

**Comment by dotlambda** [comment]
The second link sounds good to me. Let's give it a try.

**Comment by SuperSandro2000** [comment]
Changed and pushed that

**Review by dotlambda — CHANGES_REQUESTED** [review]
Should go to staging.

**Comment by dotlambda** [comment]
I guess we have to vendor the patch after all:
```
building '/nix/store/972a323a17vq8dxphk0yfvvw4sdark7n-D545.diff.drv'...

trying https://files.gnupg.net/file/data/rbaxpebkyfbchg73llt2/PHID-FILE-bmkoy3pj2jjjwf7w6l54/D545.diff
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download D545.diff from any mirror
```

**Comment by SuperSandro2000** [comment]
Okay, I changed it back

**Comment by SuperSandro2000** [comment]
Can we merge this?

File: pkgs/development/libraries/gpgme/default.nix [hunk 1]
```diff
@@ -22,28 +20,34 @@
...
   ];
 
-  outputs = [ "out" "dev" "info" ];
-
-  outputBin = "dev"; # gpgme-config; not so sure about gpgme-tool
+  postPatch = ''
+    # autoconf's beta detection requires a git repo to work
+    # and otherwise appends -unknown to the version number used in the python package which pip stumbles upon
+    substituteInPlace autogen.sh \
+      --replace 'tmp="-unknown"' 'tmp=""'
```
Comment (dotlambda) [inline pkgs/development/libraries/gpgme/default.nix:49]:
```suggestion
      --replace-fail 'tmp="-unknown"' 'tmp=""'
```

**Comment by dotlambda** [comment]
Actually https://github.com/NixOS/nixpkgs/pull/281169 marked `python312Packages.gpgme` as broken.

**Comment by SuperSandro2000** [comment]
Yeah I know but we need this change anyway and leaving it here in limbo is also not great. I can try to find it in the dependency chain and try to avoid it.

Technically pbr is only broken for the functions that use distutils.

**Pull request merged**